internal/host/validator.go

```diff
@@ -691,103 +605,41 @@ func getRealHostname(host *models.Host, inventory *models.Inventory) string {
-	switch status {
-	case ValidationSuccess:
-		return fmt.Sprintf("Hostname %s is allowed", getRealHostname(c.host, c.inventory))
-	case ValidationFailure:
 		if funk.ContainsString(hostutil.ForbiddenHostnames, getRealHostname(c.host, c.inventory)) {
-			return fmt.Sprintf("The host name %s is forbidden", getRealHostname(c.host, c.inventory))
+			return ValidationFailure, fmt.Sprintf("The host name %s is forbidden", getRealHostname(c.host, c.inventory))
 		}
-		return fmt.Sprintf("Hostname %s is forbidden, hostname should match pattern %s", getRealHostname(c.host, c.inventory), hostutil.HostnamePattern)
-	case ValidationPending:
-		return "Missing inventory"
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+		return ValidationFailure, fmt.Sprintf("Hostname %s is forbidden, hostname should match pattern %s", getRealHostname(c.host, c.inventory), hostutil.HostnamePattern)
 	}
+	return ValidationSuccess, fmt.Sprintf("Hostname %s is allowed", getRealHostname(c.host, c.inventory))
 }
 
-func (v *validator) isIgnitionDownloadable(c *validationContext) ValidationStatus {
+func (v *validator) isIgnitionDownloadable(c *validationContext) (ValidationStatus, string) {
 	if c.infraEnv != nil {
-		return ValidationSuccessSuppressOutput
+		return ValidationSuccessSuppressOutput, ""
 	}
-	if !hostutil.IsDay2Host(c.host) {
-		return ValidationSuccessSuppressOutput
+	if !hostutil.IsDay2Host(c.host) || swag.BoolValue(c.cluster.UserManagedNetworking) {
+		return ValidationSuccessSuppressOutput, ""
 	}
 	if c.host.APIVipConnectivity == "" {
-		return ValidationPending
+		return ValidationPending, "Ignition is not yet available, pending API connectivity"
 	}
 	var response models.APIVipConnectivityResponse
 	if err := json.Unmarshal([]byte(c.host.APIVipConnectivity), &response); err != nil {
-		return ValidationFailure
+		return ValidationFailure, "Ignition is not downloadable. Please ensure host connectivity to the cluster's API"
 	}
-
-	return boolValue(response.IsSuccess)
-}
-
-func (v *validator) printIgnitionDownloadable(c *validationContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
+	if boolValue(response.IsSuccess) == ValidationSuccess {
 		if swag.BoolValue(c.cluster.UserManagedNetworking) {
-			return "No API connectivity needed: User Managed Networking"
+			return ValidationSuccess, "No API connectivity needed: User Managed Networking"
 		}
-		return "Ignition is downloadable"
-	case ValidationFailure:
-		return "Ignition is not downloadable. Please ensure host connectivity to the cluster's API"
-	case ValidationPending:
-		return "Ignition is not yet available, pending API connectivity"
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+		return ValidationSuccess, "Ignition is downloadable"
 	}
+	return ValidationFailure, "Ignition is not downloadable. Please ensure host connectivity to the cluster's API"
```

Comment:
Since this message is in two places maybe it's worth defining it as a variable in the function?